lib/model/content.dart

```diff
@@ -1599,6 +1584,30 @@ class _ZulipContentParser {
     for (final node in nodes) {
       if (node is dom.Text && (node.text == '\n')) continue;
 
+      // Oddly, the way a math block gets encoded in Zulip HTML is inside a <p>.
+      if (node case dom.Element(localName: 'p', className: '', nodes: [
+            dom.Element(
+              localName: 'span',
+              className: 'katex-display') && final child, ...])) {
+        if (node.nodes case [_]
+                         || [_, dom.Element(localName: 'br'),
+                                dom.Text(text: "\n")]) {
+          // This might be too specific; we'll find out when we do #190.
+          // The case with the `<br>\n` can happen when at the end of a quote;
+          // it seems like a glitch in the server's Markdown processing,
+          // so hopefully there just aren't any further such glitches.
+          if (currentParagraph.isNotEmpty) consumeParagraph();
+          if (imageNodes.isNotEmpty) consumeImageNodes();
+          result.add(parseMathBlock(child));
+          continue;
```

Comment:
nit: How about moving the two "if not empty, consume" lines above this if/else, and the `continue` below it?

That would deduplicate them between the `if` and the `else` cases, showing that those parts don't vary. And it would also reduce the diff in the second commit.